tests/functionality.py

```diff
@@ -150,25 +139,47 @@ def testFunctionalityWithRealDatabase(self):
             # Not Mocked At All
             'Phabricator': PhabricatorProvider,
         }
-        u = Updatebot(configs, providers)
+
+    def testAllNewJobs(self):
+        # Setup
+        try_revision = "try_rev"  # Doesn't matter
+        expected_values = DEFAULT_EXPECTED_VALUES(try_revision)
+        self.configs['Bugzilla']['filed_bug_id'] = expected_values.filed_bug_id
+
+        command_mappings = {
+            "./mach vendor": expected_values.library_version_id,
+            "./mach try auto": TRY_OUTPUT(try_revision),
+            "arc diff --verbatim": ARC_OUTPUT
+        }
+        self.configs['Command']['test_mappings'] = command_mappings
+
+        # Make it
+        u = Updatebot(self.configs, self.providers)
+
+        # Ensure we don't have a dirty database with existing jobs
```

Comment:
Is there a unittest method we can define that will run before/after each test that we could move database cleanup to?

Something similar to mocha's [beforeEach/afterEach](https://mochajs.org/#asynchronous-hooks)

Reply:
I noodled around on this a bit, wrote a patch, threw it out, wrote a SO question, threw it out, and finally came up with something I think is an improvement.  But it'll be a real bitch to rebase into here, so I'd like to leave it to the end of the next patchset where I add like 4 more tests to this file.